.github/workflows/release.yml

```diff
@@ -5,141 +5,246 @@ on:
     steps:
       - uses: actions/checkout@v2
 
+      - uses: actions/download-artifact@v2
+        with:
+          name: darwinia-artifact
+          path: shared
+
       - name: Docker login
         uses: docker/login-action@v1
         with:
           username: ${{ secrets.QUAY_IO_BOT_USERNAME }}
           password: ${{ secrets.QUAY_IO_BOT_PASSWORD }}
           registry: quay.io
 
-      - name: Download deploy
-        uses: actions/download-artifact@v2
+      - uses: olegtarasov/get-tag@v2.1
+        id: tag-name
+      - uses: benjlevesque/short-sha@v1.2
+        id: short-sha
         with:
-          name: darwinia-artifact
-          path: deploy
+          length: 7
 
-      - name: Setup image
+      - name: Build and publish docker image
         env:
-          REGISTRY_HOST: quay.io
+          SHA: ${{ env.SHA }}
         run: |
-          .maintain/setup-image.sh
+          chmod +x shared/darwinia
 
-      - uses: 8398a7/action-slack@v3
-        if: failure()
-        with:
-          status: ${{ job.status }}
-          fields: repo,author,eventName,workflow,ref,commit
-        env:
-          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_INCOMING_WEBHOOK_URL }}
+          IMAGE_NAME=darwinia
+          IMAGE_PREFIX=quay.io/darwinia-network
+          IMAGE_PUSH_NAME_TAG=${IMAGE_PREFIX}/${IMAGE_NAME}:${GIT_TAG_NAME}
+          IMAGE_PUSH_NAME_SHA=${IMAGE_PREFIX}/${IMAGE_NAME}:sha-${SHA}
 
+          docker build \
+            -t ${IMAGE_NAME} \
+            -f .maintain/docker/Dockerfile \
+            .
+
+          docker tag ${IMAGE_NAME} ${IMAGE_PUSH_NAME_TAG}
+          docker tag ${IMAGE_NAME} ${IMAGE_PUSH_NAME_SHA}
+
+          docker push ${IMAGE_PUSH_NAME_TAG}
+          docker push ${IMAGE_PUSH_NAME_SHA}
 
   complete-build:
     name: Complete build
     runs-on: ubuntu-latest
-    needs: [build-image, build-wasm]
+    needs: [build-subwasm, build-image]
     steps:
       - uses: actions/checkout@v2
 
       - uses: actions/download-artifact@v2
         with:
           name: darwinia-artifact
-          path: deploy
+          path: shared
+
+      - name: Prepare deploy folder
+        run: |
+          chmod +x shared/darwinia
 
-      - name: Complete build
-        run: .maintain/complete-build.sh
+          mkdir deploy
+          cd shared
+          tar cf darwinia.tar.zst darwinia -I pzstd
+          mv darwinia.tar.zst ../deploy/
 
-      - name: Publish github release
-        uses: docker://antonyurchenko/git-release:latest
-        if: startsWith(github.ref, 'refs/tags/v')
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          DRAFT_RELEASE: "false"
-          PRE_RELEASE: "false"
-          CHANGELOG_FILE: "CHANGELOG.md"
-          ALLOW_EMPTY_CHANGELOG: "false"
+          mv crab-runtime/*.compact.wasm ../deploy/
+          mv crab-runtime/*.compact.json ../deploy/
+          mv darwinia-runtime/*.compact.wasm ../deploy/
+          mv darwinia-runtime/*.compact.json ../deploy/
```

Comment:
I don't recommend using `cd` in CI scripts (especially having `../` together), as it might cause people confused about the context and working directory. For example:

- Other developers might not notice the `cd` above while adding more filesystem commands.
- It's hard to know "where I am" and "where is this file going" when I saw `../deploy/` only.

Instead, I personally would use something like (not tested):

```
mkdir deploy
tar cf darwinia.tar.zst -C shared darwinia -I pzstd
mv darwinia.tar.zst deploy/
mv shared/crab-runtime/*.compact.{wasm,json} deploy/
```

Also, there's a small sugar `*.compact.{wasm,json}`. I forget whether that's a zsh or bash feature. Confirm before you use this.